src/Data/List/Lazy.purs

```diff
@@ -629,10 +629,7 @@ zip = zipWith Tuple
 --------------------------------------------------------------------------------
 
 instance showList :: (Show a) => Show (List a) where
-  show xs = "fromStrict (" ++ go (step xs) ++ ")"
-    where
-    go Nil = "Nil"
-    go (Cons x xs) = "Cons (" ++ show x ++ ") (" ++ go (step xs) ++ ")"
+  show xs = "toList " <> show (fromList xs :: Array a)
```

Comment:
Perhaps this is a good time to add a `take` with an ellipsis here to avoid trying to print infinite lazy lists?